packages/ckeditor5-table/src/tableutils.js

```diff
@@ -712,7 +732,8 @@ function createEmptyRows( writer, table, insertAt, rows, tableCellToInsert, attr
 // @param {module:engine/model/position~Position} insertPosition
 function createCells( cells, writer, insertPosition, attributes = {} ) {
 	for ( let i = 0; i < cells; i++ ) {
-		createEmptyTableCell( writer, insertPosition, attributes );
+		const cellAttributes = Array.isArray( attributes ) ? attributes[ attributes.length - i - 1 ] : attributes;
```

Comment:
I tried to follow the code for a while and I couldn't found out whit this is in reverse order. I would expect `attributes[ i ]` here.

But I wonder if we couldn't change implementation a bit as it is very complicated now. From the tests we copy table structure only in some edge cases and in most this array is `[ null, null, null ]`. This makes me think that maybe we do too much here and maybe on/off switch would be sufficient.

What I think now is that the default behaviour should be the one described in the ticket. Passing flag (like `useSingleColumns`) could enable the old behavior.

Let's discuss this later on as I'm not sure which way would be better and it may be better to think on it for a while rather re-implement over and over :)

Reply:
As discussed f2f: if we implement flag for old behavior it would be a breaking change and still we would need to indicate which row structure we want to copy (since above/below is resolved to `at`command level).

Reply:
The `attributes` were applied in reverse order because the table cells was inserted in reverse order (they all was inserted at the same position). This was changed in [399356e](https://github.com/ckeditor/ckeditor5/pull/6748/commits/399356e33a26c979820d7e950985e311bcf5fbcf).